### PR TITLE
Adding tail customization via sprite accessories.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- Adds a (currently unused) tail sprite accessory category that overrides tail appearance.
- Modifies some code to support tail hiding and tail colours.

## TODO
- [x] Get two-tone tails to work.
- [x] Get two-tone tails to save.
- [x] Work out why tails have disappeared from Unathi and Taj.
- [x] Check if two-tone tail icons work with animations.
- [x] Get hiding of accessory categories to work.
- [x] Get whitelisting to stop removing accessories during load.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Will depend on #4286 for two-tone tails.

File: code/_helpers/lists.dm [hunk 1]
```diff
@@ -856,3 +856,21 @@ var/global/list/json_cache = list()
 /// Is this a dense (all keys have non-null values) associative list with at least one entry?
 /proc/is_dense_assoc(var/list/L)
 	return length(L) > 0 && !isnull(L[L[1]])
+
+// I have a feeling this is a redundant proc but I couldn't find another implementation.
+/proc/lists_are_equivalent(list/A, list/B, associative)
```
Comment (out-of-phaze) [inline code/_helpers/lists.dm]: `/proc/same_entries` on line 209 of this file

**Review by out-of-phaze — DISMISSED** [review]

File: code/_helpers/lists.dm (hunk 1 above)
Comment (out-of-phaze) [reply code/_helpers/lists.dm]: actually that doesn't properly handle the recursive case, so it should prboably be edited to do `(islist(first[entry]) && islist(second[entry])) ? same_entries(first[entry], second[entry]) : first[entry == second[entry]` or something

File: code/_helpers/lists.dm [hunk 2]
```diff
@@ -205,14 +205,25 @@
...
 	if(length(first) != length(second))
 		return 0
 	for(var/entry in first)
-		if(!(entry in second) || (first[entry] != second[entry]))
+		if(!(entry in second))
+			return 0
+		var/first_entry = first[entry]
+		if(islist(first_entry))
+			var/second_entry = second[entry]
+			if(!islist(second_entry))
```
Comment (out-of-phaze) [inline code/_helpers/lists.dm:222]: technically the first line of same_entries should handle this but it's fine

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**